v6_MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,226 @@
+DropInRequest request = new DropInRequest();
+request.setCollectDeviceData(true);
+request.setMaskCardNumber(true);
+```
+
+Kotlin:
+```kotlin
+val request = DropInRequest()
+request.collectDeviceData = true
+request.maskCardNumber = true
+```
+
+See the [Braintree Android v4 Migration Guide](https://github.com/braintree/braintree_android/blob/master/v4_MIGRATION_GUIDE.md) for changes to request objects for Google Pay, Venmo, PayPal, and 3DS. 
+
+## DropInRequest
+
+The getters and setters on `DropInRequest` have been renamed with a consistent get/set pattern to allow for Kotlin synthesized properties. 
+See the examples below for a full list of optional parameters:
+
+Java:
+```java
+    DropInRequest dropInRequest = new DropInRequest();
+    dropInRequest.setShouldCollectDeviceData(true);
+    dropInRequest.setGooglePayRequest(googlePayRequest);
+    dropInRequest.setGooglePayDisabled(true);
+    dropInRequest.setPayPalRequest(paypalRequest);
+    dropInRequest.setPayPalDisabled(true);
+    dropInRequest.setVenmoDisabled(true);
+    dropInRequest.setCardDisabled(true);
+    dropInRequest.setShouldRequestThreeDSecureVerification(true);
+    dropInRequest.setThreeDSecureRequest(threeDSecureRequest);
+    dropInRequest.setShouldMaskCardNumber(true);
+    dropInRequest.setShouldMaskSecurityCode(true);
+    dropInRequest.setEnableVaultManager(true);
+    dropInRequest.setAllowVaultCardOverride(true);
+    dropInRequest.setVaultCardDefaultValue(true);
+    dropInRequest.setCardholderNameStatus(CardForm.FIELD_OPTIONAL);
+    dropInRequest.setVaultVenmoDefaultValue(true);
+```
+
+Kotlin:
+```kotlin
+    val dropInRequest = DropInRequest()
+    dropInRequest.shouldCollectDeviceData = true
+    dropInRequest.googlePayRequest = googlePayRequest
+    dropInRequest.googlePayDisabled = true
+    dropInRequest.payPalRequest = paypalRequest
+    dropInRequest.payPalDisabled = true
+    dropInRequest.venmoDisabled = true
+    dropInRequest.cardDisabled = true
+    dropInRequest.shouldRequestThreeDSecureVerification = true
+    dropInRequest.threeDSecureRequest = threeDSecureRequest
+    dropInRequest.shouldMaskCardNumber = true
+    dropInRequest.shouldMaskSecurityCode = true
+    dropInRequest.enableVaultManager = true
+    dropInRequest.allowVaultCardOverride = true
+    dropInRequest.vaultCardDefaultValue = true
+    dropInRequest.cardholderNameStatus = FIELD_OPTIONAL
+    dropInRequest.vaultVenmoDefaultValue = true
+```
+
+The full list of changed parameters is below:
+1. `clientTokent` -> removed
+1. `tokenizationKey` -> removed
+1. `amount` -> removed
+1. `intent` -> removed
+1. `collectDeviceData` -> `shouldCollectDeviceData`
+1. `googlePaymentRequest` -> `googlePayRequest`
+1. `disableGooglePayment` -> `googlePayDisabled`
+1. `paypalRequest` -> `payPalRequest`
+1. `disablePayPal` -> `payPalDisabled`
+1. `disableVenmo` -> `venmoDisabled`
+1. `disableCard` -> `cardDisabled`
+1. `requestThreeDSecureVerification` -> `shouldRequestThreeDSecureVerification`
```

Comment:
I think `should` and `Request` is redundant

Reply:
I do like the shorter name. Also indifferent here.